pkg/morph/client/client.go

```diff
@@ -54,9 +54,9 @@ type Client struct {
 
 	cfg cfg
 
-	endpoints *endpoints
+	endpoints endpoints
 
-	// switching between rpc endpoint lock
+	// switchLock protects endpoints, inactive, and subscription-related fields.
```

Comment:
is that change necessary? my idea was to clarify that mutex locks everything (no new RPCs until the switching process is finished not only protecting the fields). could you add that idea to the comment?

Reply:
Yes, because it protects shared data. Now it is obvious you can't use any of these fields without holding a mutex.
Added a comment about switching process.